crates/languages/src/python.rs

```diff
@@ -240,11 +245,120 @@ impl ContextProvider for PythonContextProvider {
         _: Option<&HashMap<String, String>>,
         _cx: &mut gpui::AppContext,
     ) -> Result<task::TaskVariables> {
+        let test_runner = TestRunner::PYTEST; // TODO: Get this from settings.
+
+        let test_target = match test_runner {
+            TestRunner::UNITTEST => self.build_unittest_target(variables)?,
+            TestRunner::PYTEST => self.build_pytest_target(variables)?,
+        };
+
+        Ok(task::TaskVariables::from_iter([test_target]))
+    }
+
+    fn associated_tasks(
+        &self,
+        _: Option<Arc<dyn language::File>>,
+        _cx: &AppContext,
+    ) -> Option<TaskTemplates> {
+        let python_path = ".venv/bin/python"; // TODO: Re-use terminal detect_venv logic for this. (How?)
+        let test_runner = TestRunner::PYTEST; // TODO: Get this from settings. (How?)
```

Comment:
You can see how Rust does it:
https://github.com/zed-industries/zed/blob/695176898e809fe097b5f6ea6db5712cf1cfefe0/crates/languages/src/rust.rs#L427